server/routes/updateChain.ts

```diff
@@ -42,6 +44,14 @@ const updateChain = async (models, req: Request, res: Response, next: NextFuncti
 
   const { active, icon_url, symbol, type, name, description, website, discord, element, telegram, github, stagesEnabled, additionalStages, customDomain, snapshot } = req.body;
 
+  if (snapshot && !(/^[a-z]+\.eth/).test(snapshot)) {
+    return next(new Error(Errors.InvalidSnapshot));
+  }
+
```

Comment:
can you add these two checks to the else if chain below prior to all the setting?